cmd/helm/status.go

```diff
@@ -191,7 +191,7 @@ func (s statusPrinter) WriteTable(out io.Writer) error {
 		// Print an extra newline
 		fmt.Fprintln(out)
 
-		cfg, err := chartutil.CoalesceValues(s.release.Chart, s.release.Config)
+		cfg, err := chartutil.MergeValues(s.release.Chart, s.release.Config)
```

Comment:
This section of code is used by `helm get all`, `helm --debug install`, and possibly other places. Before this change values that were `null` were not displayed in the output. After this change they are displayed with a value of `null` in the computed values. Is this intentional?

Reply:
Imo that is what the output should explicitly show, because that is the whole purpose of this pr. Nill values should be propagated. So any result output should list that this is what is happening. Explicitly.

Reply:
This is what I was mentioning here https://github.com/helm/helm/pull/11440#issuecomment-1280070498

Basically the original behaviour of the public function `CoalesceValues` is maintained in this PR and a new public function `MergeValues` is also created which does not trim nil values from the resulting output. This is to allow those nils to persist until the last call to TrimNilValues where they will be removed.

In short:
* MergeValues is effectively a "fixed" CoalesceValues - but it leaves nulls in place and returns them
* CoalesceValues maintains its previous _potentially_ buggy behaviour but _doesn't_ return null ever - I say potentially as its only when its called in succession as helm does internally - externally this behaviour may not have been seen

I have made the assumtion in the original comment that CoalesceValues should never return null - this may not be the case and in which case we could simplify this PR and the potential annoyance of wording between Merge and Coalesce - but it is potentially changing external behaviour which this current PR fixes albeit with the aforementioned naming challenges.

Before I make additional changes its probably worth clearing up if my assumption is valid or if we can make CoalesceValues return nulls